src/main/java/school/hei/haapi/service/StudentFileService.java

```diff
@@ -0,0 +1,80 @@
+import static school.hei.haapi.service.utils.InstantUtils.Instant_now;
+import static school.hei.haapi.service.utils.ScholarshipCertificateUtils.getAcademicYearPromotion;
+import static school.hei.haapi.service.utils.ScholarshipCertificateUtils.getAcademicYearSentence;
+
+import com.lowagie.text.DocumentException;
+import java.io.ByteArrayOutputStream;
+import lombok.AllArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.thymeleaf.TemplateEngine;
+import org.thymeleaf.context.Context;
+import org.thymeleaf.templatemode.TemplateMode;
+import org.thymeleaf.templateresolver.ClassLoaderTemplateResolver;
+import org.xhtmlrenderer.pdf.ITextRenderer;
+import school.hei.haapi.model.User;
+import school.hei.haapi.model.exception.ApiException;
+import school.hei.haapi.model.exception.NotFoundException;
+import school.hei.haapi.repository.UserRepository;
+
+@Service
+@AllArgsConstructor
+public class StudentFileService {
+  private final UserRepository userRepository;
+
+  public byte[] generatePdf(String studentId, String template) {
+    Context context = loadContext(studentId);
+    String html = parseTemplateToString(context, template);
+    return renderPdf(html);
+  }
+
+  private byte[] renderPdf(String pdf) {
+    ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+    ITextRenderer renderer = new ITextRenderer();
+
+    renderer.setDocumentFromString(pdf);
+    renderer.layout();
+
+    try {
+      renderer.createPDF(outputStream);
+    } catch (DocumentException e) {
+      throw new ApiException(SERVER_EXCEPTION, e.getMessage());
+    }
+    return outputStream.toByteArray();
+  }
+
+  private Context loadContext(String studentId) {
+    Context context = new Context();
+    User student =
+        userRepository
+            .findById(studentId)
+            .orElseThrow(() -> new NotFoundException("Student not found"));
+    context.setVariable("student", student);
+    context.setVariable("now", instantToCommonDate(Instant_now));
+    context.setVariable("academic_sentence", getAcademicYearSentence(student));
+    context.setVariable("academic_promotion", getAcademicYearPromotion(student));
+    context.setVariable("birth_day", toLocalDateReadable(student.getBirthDate()));
+    return context;
+  }
+
+  private String parseTemplateToString(Context context, String template) {
+    TemplateEngine templateEngine = configureTemplate();
+    return templateEngine.process(template, context);
+  }
+
+  private TemplateEngine configureTemplate() {
+    ClassLoaderTemplateResolver resolver = new ClassLoaderTemplateResolver();
+    resolver.setPrefix("/templates/");
+    resolver.setSuffix(".html");
+    resolver.setCharacterEncoding("UTF-8");
+    resolver.setTemplateMode(TemplateMode.HTML);
+
+    TemplateEngine templateEngine = new TemplateEngine();
+    templateEngine.setTemplateResolver(resolver);
+    return templateEngine;
+  }
```

Comment:
move to class HtmlParser,
It will implement BiFunction<String, Context, String>
only the parseTemplateToString